frontend-api/src/test/java/uk/gov/di/authentication/frontendapi/lambda/SendNotificationHandlerTest.java

```diff
@@ -168,6 +168,32 @@ void shouldReturn204AndPutMessageOnQueueForAValidVerifyEmailRequest()
         verify(sessionService).save(argThat(this::isSessionWithEmailSent));
     }
 
+    @Test
+    void shouldUseExistingOtpCodeIfOneExists() {
+        usingValidSession();
+        usingValidClientSession(CLIENT_ID);
+
+        when(codeStorageService.getOtpCode(any(String.class), any(NotificationType.class)))
+                .thenReturn(Optional.of(TEST_SIX_DIGIT_CODE));
+
+        APIGatewayProxyRequestEvent event = new APIGatewayProxyRequestEvent();
+        event.setHeaders(Map.of("Session-Id", session.getSessionId()));
+        event.setBody(
+                format(
+                        "{ \"email\": \"%s\", \"notificationType\": \"%s\", \"phoneNumber\": \"%s\" }",
+                        TEST_EMAIL_ADDRESS, VERIFY_PHONE_NUMBER, TEST_PHONE_NUMBER));
+        APIGatewayProxyResponseEvent result = handler.handleRequest(event, context);
+
```

Comment:
You want to do a check here that the same OTP code is being put on the SQS queue, as that will indicate the same code is going to be sent. So something like 
`var notifyRequest =
                new NotifyRequest(TEST_PHONE_NUMBER, VERIFY_PHONE_NUMBER, TEST_SIX_DIGIT_CODE);
        verify(awsSqsClient).send(objectMapper.writeValueAsString(notifyRequest));`